### PR TITLE
add csv node and edge from R data "Patients"

### PR DESCRIPTION
**Pull request by oanesini**
Render with `Patients Process` now provided as test resources in the repository (for label position issue, see #53)

image

### CODE REVIEW

File: java/src/main/java/io/process/analytics/tools/bpmn/generator/internal/BPMNDiagramRichBuilder.java [hunk 1]
```diff
@@ -112,7 +112,7 @@ public TDefinitions build() {
         List<JAXBElement<? extends DiagramElement>> diagramElements = bpmnPlane.getDiagramElement();
 
         bpmnShapes.stream()
-                .map(s -> new JAXBElement<>(bpmnElementQName("BPMNShape"), BPMNShape.class, null, s))
+                .map(s -> )
```
Comment (tbouffard) [inline java/src/main/java/io/process/analytics/tools/bpmn/generator/internal/BPMNDiagramRichBuilder.java:115]: :warning: compilation failure here!

File: java/src/main/java/io/process/analytics/tools/bpmn/generator/App2.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+import java.io.File;
+import java.io.IOException;
+
+import io.process.analytics.tools.bpmn.generator.input.CSVtoBPMN;
+import io.process.analytics.tools.bpmn.generator.internal.BpmnInOut;
+import io.process.analytics.tools.bpmn.generator.internal.generated.model.TDefinitions;
+import lombok.extern.log4j.Log4j2;
+
+@Log4j2
+public class App2 extends App {
```
Comment (baptistemesta) [inline java/src/main/java/io/process/analytics/tools/bpmn/generator/App2.java:15]: ℹ️  we should review the App entry point

File: java/src/main/java/io/process/analytics/tools/bpmn/generator/input/CSVtoBPMN.java [hunk 3]
```diff
@@ -0,0 +1,113 @@
+package io.process.analytics.tools.bpmn.generator.input;
```
Comment (baptistemesta) [inline java/src/main/java/io/process/analytics/tools/bpmn/generator/input/CSVtoBPMN.java:1]: ℹ️  `import` would be more consistent with `export`

File: java/src/main/java/io/process/analytics/tools/bpmn/generator/model/Edge.java [hunk 4]
```diff
@@ -41,10 +41,14 @@ public static Edge edge(String id, String from, String to) {
...
     public static Edge edge(Shape from, Shape to) {
-        return new Edge(generateRandomId(), from.getId(), to.getId(), false);
+        return new Edge(generateRandomId(), copy(from.getId()), copy(to.getId()), false);
     }
     public static Edge revertedEdge(Edge original) {
         return original.toBuilder().from(original.getTo()).to(original.getFrom()).reverted(true).build();
     }
 
+    private static String copy(String s) {
+        return s + ""; // hack to ensure the returned string is equal to the provided one but doesn't have the same reference
```
Comment (baptistemesta) [inline java/src/main/java/io/process/analytics/tools/bpmn/generator/model/Edge.java:51]: ❓ Do we really need to do that?

**Comment by baptistemesta** [comment]
merging that despite the few remarks
:shipit: 🎉

**Pull request merged**